contracts/integration/LockboxAdapter.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IXERC20} from "../shared/IXERC20/IXERC20.sol";
+import {IXERC20Lockbox} from "../shared/IXERC20/IXERC20Lockbox.sol";
+import {IConnext} from "@connext/interfaces/core/IConnext.sol";
+import {IXReceiver} from "@connext/interfaces/core/IXReceiver.sol";
+
+interface IXERC20Registry {
+  function getXERC20(address erc20) external view returns (address xerc20);
+
+  function getERC20(address xerc20) external view returns (address erc20);
+
+  function getLockbox(address erc20) external view returns (address xerc20);
+}
+
+contract LockboxAdapter is IXReceiver {
+  address immutable connext;
+  address immutable registry;
+
+  // EVENTS
+  event LockBoxWithdrawFailed(bytes _lowLevelData);
+
+  // ERRORS
+  error Forwarder__is__not__Adapter(address sender);
+  error IXERC20Adapter_WithdrawFailed();
+
+  modifier onlyConnext() {
+    require(msg.sender == connext, "Can only be called by Connext");
+    _;
+  }
+
+  constructor(address _connext, address _registry) {
+    connext = _connext;
+    registry = _registry;
+  }
+
+  /// @dev Combines Lockbox deposit and xcall using native asset as relayer fee.
+  /// @param _destination The destination domain ID.
+  /// @param _to The recipient or contract address on destination.
+  /// @param _asset The address of the asset to be sent (ERC20 or native).
+  /// @param _delegate The address on destination allowed to update slippage.
+  /// @param _amount The amount of asset to bridge.
+  /// @param _slippage The maximum slippage a user is willing to take, in BPS.
+  /// @param _callData The data that will be sent to the target contract.
+  function xcall(
+    uint32 _destination,
+    address _to,
+    address _asset,
+    address _delegate,
+    uint256 _amount,
+    uint256 _slippage,
+    bytes calldata _callData
+  ) external payable returns (bytes32) {
+    require(_amount > 0, "Amount must be greater than 0");
```

Comment:
consider to use custom error instead of require

Reply:
Got it